frontend/src/util/SmartAuth.js

```diff
@@ -0,0 +1,73 @@
+import { useLocation } from "react-router-dom";
+import { Login } from "../pages/Login";
+import Cookies from 'js-cookie';
+
+export class SmartAuth {
+
+    static #citizen = null;
+
+    static async #fetchData(token) {
+        const url = `http://auth.smartcityproject.net:8080/verify`;
+        fetch(url, {
+            method: 'POST',
+            body: encodeURIComponent('code') + '=' + encodeURIComponent(token),
+            headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+        })
+            .then(response => response.json())
+            .then(data => {
+                console.log("received user data:", data);
+                this.#citizen = data;
+            })
+            .catch(error => console.error(error));
+    }
+
+    static isLoggedIn() {
+        if (this.#citizen !== null) {
+            return true;
```

Comment:
Hier ist nicht immer der User unbedingt eingeloggt. Der war das vielleicht, als du das letzte Mal SmartAuth gefragt hast, aber in der Zwischenzeit kann die Session abgelaufen oder der User sich ausgeloggt haben. Da muss nochmal geprüft werden, ob die Session immernoch läuft.